analysis/index.js

```diff
@@ -59,20 +59,26 @@ function analysis (traceEventReader, processStatReader) {
       }
     }
 
-    const issues = {
-      'delay': analyseDelay(processStatSubset, traceEventSubset),
-      'cpu': analyseCPU(processStatSubset, traceEventSubset),
-      'memory': analyseMemory(processStatSubset, traceEventSubset),
-      'handles': analyseHandles(processStatSubset, traceEventSubset)
-    }
-    const category = issueCategory(issues, {detectNoise: true})
+    const p = Promise.all([
+      analyseDelay(processStatSubset, traceEventSubset),
+      analyseCPU(processStatSubset, traceEventSubset),
+      analyseMemory(processStatSubset, traceEventSubset),
+      analyseHandles(processStatSubset, traceEventSubset)
+    ])
+    p.then(function ([delay, cpu, memory, handles]) {
+      const issues = { delay, cpu, memory, handles }
+      const category = issueCategory(issues)
 
-    result.push({
-      'interval': intervalTime,
-      'issues': issues,
-      'issueCategory': category
+      result.push({
+        'interval': intervalTime,
+        'issues': issues,
+        'issueCategory': category
+      })
+      result.push(null)
+    })
+    p.catch(function (err) {
+      result.emit('error', err)
```

Comment:
If we make the parent function `async` then that `Promise` will swallow any thrown error that is not in the try/catch. I think any integration between `async`/`await`/`Promise` and a `callback`/`eventemitter` needs to eventually use a `.catch()` and `.then()`. `hidden-markov-model-tf` uses `async`/`await` because that is what TensorFlow.js uses.

Reply:
Should we use [`util.callbackify`](https://nodejs.org/api/util.html#util_util_callbackify_original) instead?